ARCHITECTURE.md

```diff
@@ -0,0 +1,133 @@
+
+### A short glossary
+
+Here is a list of terms having a specific meaning in the context of OIDC:
+- **Resource Server**: The server hosting private resources. In our case, a Solid server. A Resource Server receives requests authenticated with an Access Token
+- **Resource Owner**: the user, who owns some private resources 
+- **OIDC issuer**: the Solid Identity Provider, which issues Access Tokens, ID tokens, and Refresh Tokens.
+- **Client**: the application the Resource Owner uses to access its resources on the Resource Server. Technically, OAuth is a delegation protocol: the Resource Owner allows the Client to interact with the Resource Server on its behalf.  
+
+## Codemap of the client library modules
+
+Most architectural specificities are found in both client libraries modules. This section will give a high-level description of the inner working of either `@inrupt/solid-client-authn-node` or `@inrupt/solid-client-authn-browser`, leaving aside anything too module-specific.
+
+### The API
+
+Most of the code for these modules is internal, and hidden from the user. The public API is located in files available directly in `packages/*/src/`, namely in the `Session.ts` file. Users are expected to build a `Session` object, and to use it to interact with the session. How users are expected to use the public API is documented in our [public documentation](https://docs.inrupt.com/developer-tools/javascript/client-libraries/tutorial/authenticate/).
+
+### The Handler pattern
+
+Important components of this library are based on the Handler design pattern: given data contained in a request, a set of classes implementing a similar API will declare whether or not they may handle said request. Handlers declare two functions: `canHandle(request)` and `handle(request)`. `canHandle` simply returns a boolean indicating the ability of the handler to handle the request, and `handle` actually processes it. All of the handlers for a given type of requests are tracked by a handler aggregator, which will have the first handle for which `canHandle` returns `true` process the request. The handler aggregator has the same API as the handlers it aggregates, and brokers the request to the underlying handlers. More on that in the Dependency Injection section.
+
+In the context of this library, a request is an API to execute some OIDC-related operation. Handlers will determine whether they can handle the request based on the options specified by the user when making the call.
+
+#### Login
+
+Logging in is the operation that goes from the Client to the Issuer. It may result in a redirection of the user to the Issuer, but that is not always the case (in the `-node` module in particular). Handlers for the login operation are located in `packages/*/src/login/oidc/oidcHandlers/*Handler.ts`.
```

Comment:
"goes from" is a bit unclear to me here. Does this mean "the Client has to get the Resource Owner to visit a webpage of the Issuer, e.g. via redirection (in a browser-based app) or by having them click a link or scan a QR code)"?